tests/performance/density-kwok.go

```diff
@@ -0,0 +1,226 @@
+			It("should sucessfully create all fake VMIS", func() {
+				By("Creating a batch of fake VMIs")
+				createFakeVMIBatchWithKWOK(kubevirtClient, vmCount)
+
+				By("Waiting for a batch of VMIs")
+				waitRunningVMI(kubevirtClient, vmCount, vmBatchStartupLimit)
+
+				collectMetrics(startTime, filepath.Join(artifactsDir, "VMI-kwok-perf-audit-results.json"))
+			})
+		})
+
+		Context(fmt.Sprintf("[KWOK] create a batch of %d running VMs", vmCount), func() {
+			It("should sucessfully create all fake VMS", func() {
+				By("Creating a batch of VMs")
+				createFakeBatchRunningVMWithKwok(kubevirtClient, vmCount)
+
+				By("Waiting for a batch of VMs")
+				waitRunningVMI(kubevirtClient, vmCount, vmBatchStartupLimit)
+
+				collectMetrics(startTime, filepath.Join(artifactsDir, "VM-kwok-perf-audit-results.json"))
+			})
+		})
+
+	})
+
+	AfterEach(func() {
+		By("Deleting fake Nodes")
+		for i := 1; i <= nodeCount; i++ {
+			nodeName := fmt.Sprintf("kwok-node-%d", i)
+			err := k8sClient.CoreV1().Nodes().Delete(context.TODO(), nodeName, metav1.DeleteOptions{})
+			if err != nil {
+				log.Fatalf("Failed to delete node %s: %v", nodeName, err)
+			}
+		}
+	})
+})
+
+func createFakeNodesWithKwok(k8sClient *kubernetes.Clientset, count int) {
+	for i := 1; i <= count; i++ {
+		nodeName := fmt.Sprintf("kwok-node-%d", i)
+		node := createFakeNode(k8sClient, nodeName)
+		_, err := k8sClient.CoreV1().Nodes().Create(context.TODO(), node, metav1.CreateOptions{})
+		if err != nil {
+			log.Fatalf("Failed to create node %s: %v", nodeName, err)
+		}
+	}
+}
+
+func createFakeNode(k8sClient *kubernetes.Clientset, nodeName string) *corev1.Node {
+	node := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nodeName,
+			Labels: map[string]string{
+				"beta.kubernetes.io/arch":       "amd64",
+				"beta.kubernetes.io/os":         "linux",
+				"kubernetes.io/arch":            "amd64",
+				"kubernetes.io/hostname":        nodeName,
+				"kubernetes.io/os":              "linux",
+				"kubernetes.io/role":            "agent",
+				"node-role.kubernetes.io/agent": "",
+				"kubevirt.io/schedulable":       "true",
+				"type":                          "kwok",
+			},
+			Annotations: map[string]string{
+				"node.alpha.kubernetes.io/ttl": "0",
+				"kwok.x-k8s.io/node":           "fake",
+			},
+		},
+		Spec: corev1.NodeSpec{
+			Taints: []corev1.Taint{
+				{
+					Key:    "kwok.x-k8s.io/node",
+					Value:  "fake",
+					Effect: "NoSchedule",
+				},
+				{
+					Key:    "CriticalAddonsOnly",
+					Effect: corev1.TaintEffectNoSchedule,
+				},
+			},
+		},
+
+		Status: corev1.NodeStatus{
+			Allocatable: corev1.ResourceList{
+				corev1.ResourceCPU:              resource.MustParse("32"),
+				corev1.ResourceMemory:           resource.MustParse("256Gi"),
+				corev1.ResourceEphemeralStorage: resource.MustParse("100Gi"),
+				corev1.ResourcePods:             resource.MustParse("110"),
+				"devices.kubevirt.io/kvm":       resource.MustParse("0"),
```

Comment:
Why `0`?

Reply:
I was playing with different numbers on my machine and forgot to change it. Never mind, will update.